Backends/CLX/port.lisp

```diff
@@ -213,6 +212,19 @@
 				    :structure-notify)
                       :map (sheet-enabled-p sheet)))
 
+;; The following function used to live in CLX/text-selection.lisp
+;; before it was replaced by the new clipboard system. The below
+;; comment was already there. -- lokedhs 2019-02-16
+;;
+;; Incredibly crappy broken unportable Latin 1 encoder which should be
+;; replaced by various implementation-specific versions.
+(flet ((latin1-code-p (x)
+	 (not (or (< x 9) (< 10 x 32) (< #x7f x #xa0) (> x 255)))))
+  (defun string-encode (string)
+    (delete-if-not #'latin1-code-p (map 'vector #'char-code string)))
+  (defun exactly-encodable-as-string-p (string)
+    (every #'latin1-code-p (map 'vector #'char-code string))))
```

Comment:
- is string-encode used anywhere? if not then please remove it.
- exactly-encodable-as-string-p is a local function to one of %realize-mirror methods, please put it in flet
- can we implement exactly-encodable-as-string-p as the following? if so, please do that. where does latin1 requirement come from?

```
(handler-case (coerce string 'base-string)
  (:error () nil)
```

alternatively (if it must be latin-1):

```
(handler-case (babel:string-to-octets string :encoding :latin-1)
  (:no-error () t)
  (error () nil))
```

Reply:
This comes from the old implementation. It was using a function that used to reside in `text-selection.lisp` (out of convenience, I think).

I'm wondering why they even call `EXACTLY-ENCODABLE-AS-STRING-P` in the first place. Why not always set the `WM_NAME` property? It definitely shouldn't hurt, right? That would allow us to eliminate this function, as well as the two `WHEN` forms.

Reply:
I decided to always set `_NET_WN_NAME`. All applications I've checked (such as Emacs or Gnome-Terminal) seems to do it, even when the name is pure ASCII.